### PR TITLE
std.hash.crc: get rid of usingnamespace

### PR DESCRIPTION
**Pull request by blblack**
This flips things around such that std/hash/crc.zig is generated by the catalog-based generation tool, and the real code that used to be in that file is moved out to std/hash/crc/impl.zig.  The generated tests are moved to std/hash/crc/test.zig.  By going this route, we eliminate the need for usingnamespace without changing anything for callers of these interfaces.  The Crc32 tests are simply added to the fixed part of the generated output and compactified a bit.

This was the second-to-last usage of usingnamespace left in std.

### CODE REVIEW

**Comment by blblack** [comment]
@mlugg ^ ping since you were working on related things recently

File: lib/std/hash/crc.zig [hunk 1]
```diff
@@ -1,285 +1,911 @@
...
-}
+const impl = @import("crc/impl.zig");
 
-pub const Polynomial = enum(u32) {
-    IEEE = 0xedb88320,
-    Castagnoli = 0x82f63b78,
-    Koopman = 0xeb31d82e,
-    _,
-};
+pub const Crc = impl.Crc;
```
Comment (andrewrk) [inline lib/std/hash/crc.zig:5]: https://ziglang.org/documentation/master/#Avoid-Redundant-Names-in-Fully-Qualified-Namespaces

```diff
-std.hash.crc.Crc
+std.hash.Crc
```

File: lib/std/hash/crc.zig [hunk 2]
```diff
@@ -1,285 +1,911 @@
...
-}
-
-test "small crc32 castagnoli" {
-    const Crc32Castagnoli = Crc32SmallWithPoly(.Castagnoli);
-
-    try testing.expect(Crc32Castagnoli.hash("") == 0x00000000);
-    try testing.expect(Crc32Castagnoli.hash("a") == 0xc1d04330);
-    try testing.expect(Crc32Castagnoli.hash("abc") == 0x364b3fb7);
-}
+pub const Crc3Gsm = Crc(u3, .{
```
Comment (andrewrk) [inline lib/std/hash/crc.zig:17]: [same problem](https://ziglang.org/documentation/master/#Avoid-Redundant-Names-in-Fully-Qualified-Namespaces)

```diff
-std.hash.crc.Crc3Gsm
+std.hash.crc.@"3Gsm"
```

or perhaps `std.hash.crc.catalog.@"3GSm"`

**Review by andrewrk — CHANGES_REQUESTED** [review]
Thanks for the patch!

File: lib/std/hash/crc.zig (hunk 1 above)
Comment (blblack) [reply lib/std/hash/crc.zig:5]: These will be interface changes beyond the relatively-pure usingnamspace refactor, to be clear.

I'm not sure I can quite do it like that without going back to `usingnamespace` or generating `hash.zig`, I think.  But I could do something like:

```
-std.hash.crc.Crc
+std.hash.crc.Create
```

**Comment by andrewrk** [comment]
I see. If this is a non-breaking change then it can be merged and the namespace problems can be solved later.

File: lib/std/hash/crc.zig (hunk 1 above)
Comment (blblack) [reply lib/std/hash/crc.zig:5]: Hmmm I'm mixing up two different concerns here.  I could for that one case, put the interface up at `+std.hash.Crc` manually.  Really, I could move all of the stuff in `std/crc/impl.zig` up to `std/hash.zig` together.

**Comment by blblack** [comment]
I have a local version now that solves the namespace problems, I think (I went with `std.hash.crc.Create` and `std.hash.crc.@"3Gsm"`).  There was one needed fixup in an std consumer, in `lib/std/compress/xz/block.zig`:

```
-const Crc64 = std.hash.crc.Crc64Xz;
+const Crc64 = std.hash.crc.@"64Xz";
```

This would of course require any 3rd parties to update similarly, for all of those esoteric catalog CRCs.  I'm fine with either way, up to you.  Can either push the rename version over this, or leave it as it currently is (just usingnamespace refactor, no name changes).

**Comment by blblack** [comment]
@andrewrk - I'm assuming you'd rather avoid the renames with 0.12 incoming, based on:

> If this is a non-breaking change then it can be merged and the namespace problems can be solved later.

[yes, it is otherwise just a non-breaking refactor to kill usingnamespace].

However, I don't think I can clear the "change requested" part, I think you or someone else with merge rights needs to move it forward as-is.

**Comment by andrewrk** [comment]
Thanks!

**Pull request merged**